### PR TITLE
Add dialog to change default backup location

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Adds dialog to allow changing default backup location from the backup panel.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-mount-picker.ts [hunk 1]
```diff
@@ -55,9 +55,10 @@ class HaMountPicker extends LitElement {
       graphic="icon"
       .value=${__BACKUP_DATA_DISK__}
     >
-      <span
-        >${this.hass.localize("ui.components.mount-picker.use_datadisk")}</span
-      >
+      <span>
+        ${this.hass.localize("ui.components.mount-picker.use_datadisk") ||
+        "Use data disk for backup"}
```
Comment (ludeeus) [inline src/components/ha-mount-picker.ts:60]: We can not use this translation from the supervisor panel, in this case I think its OK its not translated.
There are several other places in that panel that does it the same way.

File: src/components/ha-mount-picker.ts [hunk 2]
```diff
@@ -137,6 +138,7 @@ class HaMountPicker extends LitElement {
     try {
       if (isComponentLoaded(this.hass, "hassio")) {
         this._mounts = await fetchSupervisorMounts(this.hass);
+        this.value = this._mounts.default_backup_mount || __BACKUP_DATA_DISK__;
```
Comment (ludeeus) [inline src/components/ha-mount-picker.ts:141]: By setting this, we are showing whats in use now when opening using the selector.

Comment (bramkragten) [reply src/components/ha-mount-picker.ts:141]: Add a guard to only do this if usage is backup

Comment (ludeeus) [reply src/components/ha-mount-picker.ts:141]: Huh? usage would always be backup here.
If its undefined or null, it means the same, /bckup is in use

Comment (bramkragten) [reply src/components/ha-mount-picker.ts:141]: This is an input element for picking mounts, it also allows media, but is not used now. 

But if the user picked something else (like it is used in a service call, or blueprint, etc) the value could be set already and we now override that value with the default, which is wrong.
You can also move this logic to the backup location dialog.

Comment (ludeeus) [reply src/components/ha-mount-picker.ts:141]: No? its a `backup_location` selector, that always are backup

Comment (ludeeus) [reply src/components/ha-mount-picker.ts:141]: `default_backup_mount` from SU API is always also backup mount.
Everything here is backup only

Comment (ludeeus) [reply src/components/ha-mount-picker.ts:141]: Oh.... F, this is in mount picker...
Yeah...

File: src/components/ha-mount-picker.ts [hunk 3]
```diff
@@ -137,6 +138,10 @@ class HaMountPicker extends LitElement {
     try {
       if (isComponentLoaded(this.hass, "hassio")) {
         this._mounts = await fetchSupervisorMounts(this.hass);
+        if (this.usage === SupervisorMountUsage.BACKUP) {
```
Comment (bramkragten) [inline src/components/ha-mount-picker.ts:141]:
```suggestion
        if (this.usage === SupervisorMountUsage.BACKUP && !this.value) {
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**